<robot name="four_robot">

<material name="black">
        <color rgba="0 0 0 1"/>
</material>
<material name="red">
        <color rgba="0.8 0 0 1"/>
</material>
<material name="blue">
	<color rgba="0 0 0.8 1"/>
</material>
<material name="green">
	<color rgba="0 0.8 0 1"/>
</material>

<link name="base_link">
	<visual>
		<geometry>
			<box size="0.1 0.1 0.9"/>
		</geometry>
		<material name="black" />
	</visual>
</link>

<link name="link_two">
        <visual>
                <geometry>
                        <box size="0.6 0.1 0.1"/>
                </geometry>
		<origin xyz="0.0 0.0 0.0"/>
                <material name="red" />
        </visual>
</link>

<joint name="base_to_two" type="prismatic">
	<axis xyz="0 0 1" />
	<parent link="base_link"/>
	<child link="link_two"/>
	<origin xyz="0.2 0.0 0.2"/>
	<limit effort="100" velocity="100" lower="0" upper="-0.2"/>
</joint> 

<link name="link_three">
        <visual>
                <geometry>
                        <box size="0.1 0.6 0.1"/>
                </geometry>
		<origin xyz="0.0 0.0 0.0"/>
                <material name="blue" />
        </visual>
</link>

<joint name="two_to_three" type="prismatic">
        <parent link="link_two"/>
        <child link="link_three"/>
        <origin xyz="0.0 0.0 -0.1"/>
        <limit effort="100" velocity="100" lower="-0.2" upper="0.2"/>
</joint>

<link name="link_four">
        <visual>
                <geometry>
                        <box size="0.1 0.1 0.1"/>
                </geometry>
		<origin xyz="0.0 0.0 0.0"/>
                <material name="green" />
        </visual>
</link>

<joint name="three_to_four" type="prismatic">
	<axis xyz="0 1 0" />
        <parent link="link_three"/>
        <child link="link_four"/>
        <origin xyz="0.0 0.0 -0.1"/>
        <limit effort="100" velocity="100" lower="0.25" upper="-0.25"/>
</joint>

</robot>
